### ISSUE TITLE
Let CiviForm admins archive or delete programs

### ISSUE THREAD

**Issue by neel-a-mehta** [civiform/civiform#1375]
CiviForm admins can now archive questions, but they still can't archive programs as far as I can tell. We should enable archiving (and, of course, unarchiving them).

**Comment by baipai** [comment]
Done with archive and delete programs, will need to work on unarchiving one,

**Comment by neel-a-mehta** [comment]
@baipai great, feel free to post your mocks here or in slack once you're ready

**Comment by lindsayzad** [comment]
Open questions here on the design/PM side: 
1. What should the ability to archive a program look like?  What is the proper design for the archive button (on the admin's programs home page, within program details)? 
2. Should a confirmation dialog popup that shows the impact of what is being done (what does it mean for the program to be archived/deleted)? 
3. How would an admin unarchive a program? Where would they find the list or archived programs and how would they navigate to unarchiving them?
4. Should we allow archive, delete, or both?
5. What should happen to the applicant data when the program is archived/deleted? Is it still accessible by the admin to download, or by the BE in future programs with the same questions? If an admin can still access those applications, how? How can they tell the difference between an archived program and an unarchived program?
6. Validate that the archived program does not appear for applicants and there is no way for them to access it.
7. What should the user experience be if a program is archived while an applicant is in the process of filling it out?

**Comment by baipai** [comment]
@lindsayzad Just saw the questions, I will answer them asap.

**Comment by baipai** [comment]
I am not quite sure if my answer should be based on the last design we did(the one that was presented to the city but hasn't really applied to the current system yet) or based on the one on the staging?

If it is based on the last design we did, then here maybe some of the answers:

1. Here is the design to find the archive button.
![Archive](https://user-images.githubusercontent.com/11348330/130639354-d5d6e53b-8787-4e2b-9ff3-bf0e45ef671a.png)

2. Yes, there should be a success toast. I understand there are different success toast texts across different screens. I am thinking if one success toast text for all may help to align the information in a more systematic way? I am thinking “Changes successfully saved”. 
![success toast](https://user-images.githubusercontent.com/11348330/130642451-042cdb8f-7217-4a32-b392-f28a09393ee9.png)

3. There is different status on the top, users can click them to figure out the different status problem. When they click “Archived”, they will be able to unarchive the problem there.
![Programs unarchive](https://user-images.githubusercontent.com/11348330/130645021-2cccd89e-9237-4356-8a9e-b25801bd28d1.png)

**Comment by baipai** [comment]
I know the last design is very different from the staging, are the answers so far in the right direction? please let me know what is the best way to support this?

**Comment by lindsayzad** [comment]
Thanks for the designs here! It looks like there will be a lot of smaller work items in order to close this issue.  We will reach out if any questions come up

**Comment by lindsayzad** [comment]
I've broken the dev work needed here into steps below.  The delete work will be prioritized first to unblock it for current admins, and the archive flow can be tackled later as it is a more involved, bigger change.

1. Add a feature flag for delete changes
2. Add BE functionality for delete.  The program will be unrecoverable once deleted, but this should not affect the questions or their saved answers.  Those will stay in the DB, and a civiform admin would have to go and delete the questions individually
2. FE: Add a button to the program view for civiform admins for delete
3. FE: Add a warning dialogue for the delete button to the FE; tell admins they will no longer be able to access the program or any of the applications if they delete it
4. Add a delete browser test/ Ramp up/Any cleanup
5. Add a feature flag for archive changes
5. Add Archive functionality to the BE (More difficult, to be planned before implementing)
6.  Update the FE to match the mocks above
7. Add browser test coverage /Ramp up/ Any cleanup

**Comment by cassandralfernandes** [comment]
Most update mock [here](https://docs.google.com/presentation/d/15fcq3dYFgy1Dfy2UjZybmHD9KIW5cz8Q1BjwgOpLJ7g/edit#slide=id.g13e43d0b668_0_65)

**Comment by clouser2** [comment]
I definitely think this is a large item and I want to make sure that the original need is still present for the feature. Some questions around the current proposal:

* Can we clarify the distinction of what archiving a program does versus deleting it entirely?
* Questions around what to do with in-progress / previously submitted applications
  - Will program admins continue to see the applications in their dashboard?
  - Will applicants continue to see the data in their dashboard?
  - If answers to the questions from the deleted program exist in the applicant data, should we remove it as part of deleting the program?
  - Is it acceptable to delete applicant data that may have been used for a program in the past?

https://github.com/civiform/civiform/issues/1375#issuecomment-942748201 mentions that hiding the data from the index solves a large portion of the need (making sure the program isn't available to new applicants). Two downsides mentioned are:
 1) CiviForm admins / program admins still see hidden programs in their list
 2) Applicants with the program URL can still access the application

One approach to mitigating the former issue could be to 1) add an indication that a program is hidden to the top-level card and 2) sort these below visible programs by default in the list.

We could potentially mitigate the latter issue by redirecting the applicant to the program dashbaord if the program was hidden *and* they didn't have an active or draft application for the program.

Archival / deletion of programs is going to be a large lift and I think we might need to do more work to understand the concrete need and how we plan to deal with the associated implications of a feature.

@cassandralfernandes , do you mind assessing what direction we want to go here?

**Comment by seanmalbert** [comment]
@isabelcshaw and @cassandralfernandes ,
This feature resurfaced recently with Arkansas with this request:

> Can we separate or hide the active and draft programs on different tabs on the All Programs page, https://civiform-staging.ar.gov/admin/programs? There are some test programs that we would like to delete completely, but we realize that after a program is created and entries have been recorded, that deleting them would (may) not be possible. A suggestion would be to create two categories for the programs page, “All Programs” and “Hidden” and create the ability to hide programs as they become inactive or are no longer needed.

There appear to be a few different asks:
1. to break up programs by status under different tabs
2. to completely delete programs that are no longer needed
3. to hide programs from the main admin/programs view

The third ask seems in line with the last comment above starting with:
> https://github.com/civiform/civiform/issues/1375#issuecomment-942748201 mentions that hiding the data from the index solves a large portion of the need (making sure the program isn't available to new applicants).

and the ability to hide programs from the index or sorting programs marked as hidden to the bottom, gets us the farthest with the least amount of effort.

**Comment by msprenke** [comment]
What happens to the applications and questions assigned to program?

**Comment by reemaka** [comment]
+1 to the point others have made on this thread about hiding programs being the simplest version of this that also accomplishes the most of the goals here.

We already have the ability to hide programs from the applicant so that only those with a direct link can access the program. We can add a "Hide" option to the dropdown on the Programs page that sets the program to hidden from applicants and we can move these programs to the bottom. All the applications and questions will still be intact. The admin will have to publish a new version to save the changes.

I'm happy to start working on this if others are in agreement with this solution.

**Comment by reemaka** [comment]
Got it, makes sense. Allowing deleting programs with confirmation might be the best option then. Going to discuss among the team to hopefully decide on a path forward.

**Comment by msprenke** [comment]
Duplicate item.  Consolidated to https://github.com/civiform/civiform/issues/5806
